### PR TITLE
feat: optional minio + add s3 support with localstack

### PR DESCRIPTION
**Pull request by kaanyagci**
## Description

Closes FL-567

## Checklist

- [ ] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: charts/substra-backend/values.yaml [hunk 1]
```diff
@@ -856,12 +856,19 @@ docker-registry:
 minio:
   enabled: false
   auth:
-    rootUser: minio
-    rootPassword: minio1234
+    rootUser: &objecStoreRootUser minio
+    rootPassword: &objectStoreRootPassword minio1234
```
Comment (SdgJlbl) [inline charts/substra-backend/values.yaml:860]: How does that works? it concatenates the content of the env var + `minio`, or is `minio` the default value if the env var is not set?

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: charts/substra-backend/values.yaml (hunk 1 above)
Comment (kaanyagci) [reply charts/substra-backend/values.yaml:860]: It creates two re-usable variables inside of the `values.yaml` file `objecStoreRootUser ` and `objectStoreRootPassword ` with values `minio ` and `minio1234`. I was creates those variables to test if the CI issue is related to the missing environment variables and it seems like it, I'll cleanup those changes to make it re-usable for chart users as well :)

Comment (SdgJlbl) [reply charts/substra-backend/values.yaml:860]: Thanks for the explanation, very clear 👍

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by ThibaultFy — APPROVED** [review]
LGTM :)

**Pull request merged**